x-pack/plugins/fleet/server/services/package_policy.test.ts

```diff
@@ -318,4 +324,181 @@ describe('Package policy service', () => {
+      const callbackA: ExternalCallback[1] = jest.fn(async (ds) => {
+        callbackCallingOrder.push('a');
+        return ds;
+      });
+
+      const callbackB: ExternalCallback[1] = jest.fn(async (ds) => {
+        callbackCallingOrder.push('b');
+        return ds;
+      });
+
+      appContextService.addExternalCallback('packagePolicyCreate', callbackA);
+      appContextService.addExternalCallback('packagePolicyCreate', callbackB);
+
+      await packagePolicyService.runExternalCallbacks(
+        'packagePolicyCreate',
+        NewPackagePolicySchema,
+        newPackagePolicy,
+        context,
+        request
+      );
+      expect(callbackCallingOrder).toEqual(['a', 'b']);
+    });
+
+    it('should feed package policy returned by last callback', async () => {
+      appContextService.addExternalCallback('packagePolicyCreate', callbackOne);
+      appContextService.addExternalCallback('packagePolicyCreate', callbackTwo);
+
+      await packagePolicyService.runExternalCallbacks(
+        'packagePolicyCreate',
+        NewPackagePolicySchema,
+        newPackagePolicy,
+        context,
+        request
+      );
+
+      expect((callbackOne as jest.Mock).mock.calls[0][0].inputs).toHaveLength(0);
+      expect((callbackTwo as jest.Mock).mock.calls[0][0].inputs).toHaveLength(1);
+      expect((callbackTwo as jest.Mock).mock.calls[0][0].inputs[0].config.one.value).toEqual(
+        'inserted by callbackOne'
+      );
+    });
+
+    describe('with a callback that throws an exception', () => {
+      const callbackThree: ExternalCallback[1] = jest.fn(async () => {
+        callbackCallingOrder.push('three');
+        throw new Error('callbackThree threw error on purpose');
+      });
+
+      const callbackFour: ExternalCallback[1] = jest.fn(async (ds) => {
+        callbackCallingOrder.push('four');
+        return {
+          ...ds,
+          inputs: [
+            {
+              ...ds.inputs[0],
+              config: {
+                ...ds.inputs[0].config,
+                four: {
+                  value: 'inserted by callbackFour',
+                },
+              },
+            },
+          ],
+        };
+      });
+
+      beforeEach(() => {
+        appContextService.addExternalCallback('packagePolicyCreate', callbackOne);
+        appContextService.addExternalCallback('packagePolicyCreate', callbackTwo);
+        appContextService.addExternalCallback('packagePolicyCreate', callbackThree);
+        appContextService.addExternalCallback('packagePolicyCreate', callbackFour);
+      });
+
+      it('should fail to execute remaining callbacks after a callback exception', async () => {
+        try {
+          await packagePolicyService.runExternalCallbacks(
+            'packagePolicyCreate',
+            NewPackagePolicySchema,
+            newPackagePolicy,
+            context,
+            request
+          );
+        } catch (e) {
+          // expecting an error
```

Comment:
We can probably add the rejection itself to our expectations: https://www.valentinog.com/blog/throw-async/#how-to-throw-errors-from-async-functions-in-javascript-catch-me-if-you-can

Reply:
do you mean like in the test after it?

Reply:
you have a comment that we expect an error. But no code `expect()` ing that error.

Here are a few more patterns for how to verify that `e`: https://jestjs.io/docs/en/asynchronous#asyncawait

Reply:
My bad on all this. _the next test_ verifies the error/rejection as Candace pointed out. This one is *only* verifying that callbacks stop. Which is good.